src/sql-parser/src/parser.rs

```diff
@@ -322,16 +322,25 @@ impl<'a> Parser<'a> {
                     .map_no_statement_parser_err();
             }
 
-            let statement = self.parse_statement()?;
-            stmts.push(statement);
+            let (statement, sql) = self.parse_statement()?;
+            stmts.push((statement, sql));
             expecting_statement_delimiter = true;
         }
         Ok(stmts)
     }
+    /// Parse a single top-level statement (such as SELECT, INSERT, CREATE, etc.),
+    /// stopping before the statement separator, if any. Returns the parsed statement and the SQL
+    /// fragment corresponding to it.
+    fn parse_statement(&mut self) -> Result<(Statement<Raw>, &'a str), ParserStatementError> {
```

Comment:
The fact that nearly every caller of this function and its relatives ignores the second return value here is a strong signal that this is not the right API design. ISTM we should either have a separate set of functions that return the raw string alongside the AST node, or we should return a named struct.

Reply:
Strong preference for *not* adding different functions, because I wanted to force myself (and really, anyone calling this) to think "should I be logging this SQL?"

However, I'm fine with the idea of making it a named struct! I'll go ahead and make that change.